// Copyright 2025 Schaeffler Monitoring Services GmbH
//
// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
// documentation files(the “Software”), to deal in the Software without restriction, including without limitation the 
// rights to use, copy, modify, merge, publish, distribute, sublicense, and /or sell copies of the Software, and to 
// permit persons to whom the Software is furnished to do so, subject to the following conditions :
//
// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the 
// Software.
//
// THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE 
// WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE AUTHORS OR 
// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR 
// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

syntax = "proto3";

option optimize_for = LITE_RUNTIME;
package smartcheck;

import "SystemState.proto";
import "DeviceConfig.proto";
import "JobConfig.proto";
import "JobData.proto";
import "TimeSignal.proto";
import "Trend.proto";
import "ClassificationData.proto";

message TransferMessage {
	oneof content {
		SystemState system_state = 10;
		DeviceConfig device_config = 20;
		JobConfig job_config = 21;
		CharacteristicValueConfig characteristic_value_config = 22;
		MeasurementConfig measurement_config = 23;
		JobData job_data = 100;
		TimeSignal timesignal = 101;
		CharacteristicValueData characteristic_value_data = 102;
		Trend trend = 103;
		ClassificationData classification_data = 104;
	}
}
